### PR TITLE
HW_7 done

### PR DESCRIPTION
**Pull request by dubinchuk**
Please review hw_7

### CODE REVIEW

File: HW_7/task 7.2.3.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        vowelsCounter.push(counter);
+        });
+
+    const sortIndex = [...vowelsCounter.keys()].sort((a, b) => vowelsCounter[a] - vowelsCounter[b]);
+    const result = [];
+    sortIndex.map(el => result.push(array[el]));
+    return result;
+}
+
+console.log(countVowels(arr));
```
Comment (anatoly-karpovich) [inline HW_7/task 7.2.3.js:23]:
```suggestion
function countVowels(word = "") {
  return word.match(/[aeiou]/gi).length;
}

function sortByVowels(arr) {
  return arr.toSorted((a, b) => countVowels(a) - countVowels(b));
}
console.log(sortByVowels(["aaa", "aa", "a"]));
```

File: HW_7/task 7.2.4.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+console.log(checkForPair(array))
+
+//   Усложнение: в массиве могут быть вложены еще массивы на разной глубине. (ПОПРОБУЙТЕ МЕТОД .flat(), изучите как он работает с разными глубинами вложенности)
+//   Вернуть нужно всё также есть ли у каждой скобочки соответствующая пара. 
+//   Пример:
+//   const arr = [[['(']], ')', '(', ')', ')', ['(', ['('], [')']]]
+
+const array2 = [[['(']], ')', '(', ')', ')', ['(', ['('], [')']]];
+
+const checkForPair2 = (arr) => arr.flat(Infinity).filter(el => el === '(').length === arr.flat(Infinity).filter(el => el === ')').length;
```
Comment (anatoly-karpovich) [inline HW_7/task 7.2.4.js:17]: Good! Just one moment: you are performing flat(infinity) 2 times. Better to flat before filtering. Yes, function won't be a one-liner, but..))

**Review by anatoly-karpovich — APPROVED** [review]
Well done!

**Review by VladimirLQA — APPROVED** [review]

File: HW_7/task 7.2.4.js (hunk 2 above)
Comment (dubinchuk) [reply HW_7/task 7.2.4.js:17]: Thanks, I'll keep in mind for the future that the code will work faster in this case

File: HW_7/task 7.2.3.js (hunk 1 above)
Comment (dubinchuk) [reply HW_7/task 7.2.3.js:23]: This code is much more optimized than mine! I'll keep my solution in repository and save yours as etalon

**Pull request merged**